presto-main/src/main/java/io/prestosql/metadata/Metadata.java

```diff
@@ -571,4 +573,62 @@ default ResolvedFunction getCoercion(Type fromType, Type toType)
      * The method is used by the engine to determine if a materialized view is current with respect to the tables it depends on.
      */
     MaterializedViewFreshness getMaterializedViewFreshness(Session session, TableHandle tableHandle);
+
+    /**
+     * Redirect to another table or view. Returns the original table name if the redirection doesn't happen.
+     */
+    default QualifiedObjectName redirectTable(Session session, QualifiedObjectName tableName, WarningCollector warningCollector)
+    {
+        return tableName;
+    }
+
+    /**
+     * Redirect to another pair of source and target names. Returns original table names if if the redirection doesn't happen.
+     */
+    default RenameRedirection redirectTableRename(Session session, QualifiedObjectName sourceTableName, QualifiedObjectName targetTableName, WarningCollector warningCollector)
+    {
+        return new RenameRedirection(sourceTableName, targetTableName);
```

Comment:
This gets overridden, so please remove the default implementation, can be misleading.